gtsam/hybrid/DCMixtureFactor.h

```diff
@@ -140,22 +159,25 @@ class DCMixtureFactor : public DCFactor {
   GaussianFactor::shared_ptr linearize(
       const Values& continuousVals,
       const DiscreteValues& discreteVals) const override {
-    // Retrieve the assignment to our discrete key.
-    const size_t assignment = discreteVals.at(dk_.first);
-
-    // `assignment` indexes the nonlinear factors we have stored to compute the
-    // error.
-    return factors_[assignment].linearize(continuousVals);
+    auto factor = factors_(discreteVals);
+    return factor->linearize(continuousVals);
   }
 
   /// Linearize all the continuous factors to get a DCGaussianMixtureFactor.
   DCFactor::shared_ptr linearize(const Values& continuousVals) const override {
-    std::vector<GaussianFactor::shared_ptr> linearized_factors;
-    for (size_t i = 0; i < factors_.size(); i++) {
-      auto linearized = factors_[i].linearize(continuousVals);
-      linearized_factors.push_back(linearized);
-    }
-    return boost::make_shared<DCGaussianMixtureFactor>(keys_, dk_, linearized_factors);
+    std::function<GaussianFactor::shared_ptr(
```

Comment:
auto?

Reply:
Sadly, I already tried using `auto` but the compiler is having some problems with that. :(